orbslam_yaml.go

```diff
@@ -0,0 +1,258 @@
+	props, err := cam.Properties(ctx)
+	if err != nil {
+		return err
+	}
+	if props.IntrinsicParams == nil {
+		return transform.NewNoIntrinsicsError("Intrinsics do not exist")
+	}
+	if err = props.IntrinsicParams.CheckValid(); err != nil {
+		return err
+	}
+	if props.DistortionParams == nil {
+		return transform.NewNoIntrinsicsError("Distortion parameters do not exist")
+	}
+	// create orbslam struct to generate yaml file with
+	var cameraModel transform.PinholeCameraModel
+	cameraModel.PinholeCameraIntrinsics = props.IntrinsicParams
+
+	if props.DistortionParams != nil {
+		cameraModel.Distortion = props.DistortionParams
+	}
+	orbslam, err := orbSvc.orbCamMaker(&cameraModel)
+	if err != nil {
+		return err
+	}
+
+	// Check for maps in the specified directory and add map to yaml config
+	loadMapTimeStamp, loadMapName, err := orbSvc.checkMaps()
+	if err != nil {
+		orbSvc.logger.Debugf("Error occurred while parsing %s for maps, building map from scratch", orbSvc.dataDirectory)
+	}
+	if loadMapTimeStamp == "" {
+		loadMapTimeStamp = time.Now().UTC().Format(dataprocess.SlamTimeFormat)
+	} else {
+		orbslam.LoadMapLoc = "\"" + loadMapName + "\""
+	}
+
+	// yamlFileName uses the timestamp from the loaded map if one was available
+	// this gives the option to load images into the map if they were generated at a later time
+	// orbslam also checks for the most recently generated yaml file to prevent any issues with timestamps here
+	yamlFileName := filepath.Join(orbSvc.dataDirectory, "config", orbSvc.primarySensorName+"_data_"+loadMapTimeStamp+".yaml")
+
+	// generate yaml file
+	yamlData, err := yaml.Marshal(&orbslam)
+	if err != nil {
+		return errors.Wrap(err, "Error while Marshaling YAML file")
+	}
+
+	//nolint:gosec
+	outfile, err := os.Create(yamlFileName)
+	if err != nil {
+		return err
+	}
+
+	if _, err = outfile.WriteString(yamlFilePrefixBytes); err != nil {
+		return err
+	}
+
+	if _, err = outfile.Write(yamlData); err != nil {
+		return err
+	}
+	return outfile.Close()
+}
+
+func (orbSvc *orbslamService) orbConfigToInt(key string, def int) (int, error) {
+	valStr, ok := orbSvc.configParams[key]
+	if !ok {
+		orbSvc.logger.Debugf("Parameter %s not found, using default value %d", key, def)
+		return def, nil
+	}
+
+	val, err := strconv.Atoi(valStr)
+	if err != nil {
+		return 0, errors.Errorf("Parameter %s has an invalid definition", key)
+	}
+
+	return val, nil
+}
+
+func (orbSvc *orbslamService) orbConfigToFloat(key string, def float64) (float64, error) {
+	valStr, ok := orbSvc.configParams[key]
+	if !ok {
+		orbSvc.logger.Debugf("Parameter %s not found, using default value %f", key, def)
+		return def, nil
+	}
+
+	val, err := strconv.ParseFloat(valStr, 64)
+	if err != nil {
+		return 0, errors.Errorf("Parameter %s has an invalid definition", key)
+	}
+	return val, nil
+}
+
+// Checks the map folder within the data directory for an existing map.
+// Will grab the most recently generated map, if one exists.
```

Comment:
[nit] 
```suggestion
// checkMaps checks the map folder within the data directory for an existing map.
// It will grab the most recently generated map if one exists.
```